### PR TITLE
Added HardShrink transfer operator for Keras

### PR DESCRIPTION
**Pull request by amitsrivastava78**
New operator along with test cases added to Keras

### CODE REVIEW

**Comment by Dref360** [comment]
Do you have any reference that shows that this new activation function is better in certain cases?

Also, this may be better suited for keras-contrib since I don't think this is widely spread.

**Comment by amitsrivastava78** [comment]
@nataliaponomareva , pls spend some time for the review, would really appriciate your comments, this will help me to understand better this community and contribute more.

**Comment by amitsrivastava78** [comment]
@tanzhenyu , @nataliaponomareva and @rthadur , can you pls review the PR

**Comment by amitsrivastava78** [comment]
@tanzhenyu , @nataliaponomareva and @rthadur , can you pls review the PR

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @tanzhenyu, @nataliaponomareva: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 15 days with no activity and the `awaiting review` label has been applied.

**Review by tanzhenyu — COMMENTED** [review]
Code wise looks good to me. But I think this should be submitted to here, instead of core tensorflow:
https://github.com/tensorflow/addons/tree/master/tensorflow_addons/layers/python

**Comment by amitsrivastava78** [comment]
@tanzhenyu , this operator is supported by many frameworks like Pytorch, BIGDL and paddle , this operator is used in ADMM algorithm(there are several papers on this which you can refer). There is a arxiv paper which uses this operator as well:-
https://arxiv.org/pdf/1711.00905.pdf
Also all the framework use it in the same place of softmax,relu etc, Kindly check and let me know your opinion.

**Comment by tanzhenyu** [comment]
@amitsrivastava78 Best regards, but I tend to agree with @Dref360 that it would be better to merge this once you have a reference that shows this layer is statistically better than other layers, at least in some certain use cases.

**Comment by amitsrivastava78** [comment]
@tanzhenyu , I understand your concern, kindly refer to this paper below , refer to page 13 which has some comparison results. If you still feel that this need not be contributed here, i will close the PR and follow your suggestion.

https://www.researchgate.net/publication/220660052_Diffusion-Inspired_Shrinkage_Functions_and_Stability_Results_for_Wavelet_Denoising

Regards
Amit

**Comment by tanzhenyu** [comment]
> @tanzhenyu , I understand your concern, kindly refer to this paper below , refer to page 13 which has some comparison results. If you still feel that this need not be contributed here, i will close the PR and follow your suggestion.
> 
> https://www.researchgate.net/publication/220660052_Diffusion-Inspired_Shrinkage_Functions_and_Stability_Results_for_Wavelet_Denoising
> 
> Regards
> Amit

This paper proposed a wavelet shrinkage method, and hardshrink is merely used as a benchmark for comparison purpose. I'm totally fine if you want to keep this and submit it to the add-on repo, which I think many users are relying on as well.

**Comment by amitsrivastava78** [comment]
@tanzhenyu, thanks a lot for the review and time you have spend on the PR, it was nice discussing with you. Hope to have more discussions in future. I am closing this PR as per the suggestion.

Regards
Amit

**Comment by tanzhenyu** [comment]
Thank you for your contribution!
